src/estimators/tests/bloom_filters_test.py

```diff
@@ -100,13 +112,67 @@ def test_factory(self, bloom_filter_class, kwargs):
     adbf.add([1, 1])
     self.assertEqual(np.sum(adbf.sketch), 1)
 
+  @parameterized.parameters(
+      (GeometricBloomFilter, {}),
+  )
+  def test_geobf_factory(self, bloom_filter_class, kwargs):
+    factory = bloom_filter_class.get_sketch_factory(length=2, **kwargs)
+    adbf = factory(1)
+    self.assertEqual(np.sum(adbf.sketch), 0)
+    adbf.add([1, 1])
+    self.assertEqual(np.sum(adbf.sketch), 0)
+    adbf.add([0.2, 1])
+    self.assertEqual(np.sum(adbf.sketch), 1)
+
   def test_inversion(self):
     log = invert_monotonic(math.exp, epsilon=0.1**9)
     self.assertAlmostEqual(log(1), 0)
     self.assertAlmostEqual(log(math.exp(2)), 2)
     self.assertAlmostEqual(log(math.exp(42.42)), 42.42)
 
 
+class GeometricBloomFilterTest(absltest.TestCase):
+
+  def test_insert_lookup_with_random_seed(self):
+    b = BloomFilter(length=15, random_seed=2)
+
+    self.assertNotIn(1, b)
+    b.add(1)
+    self.assertIn(1, b)
+    self.assertNotIn(2, b)
+
+  def test_factory(self):
+    factory = GeometricBloomFilter.get_sketch_factory(length=15)
+
+    b = factory(2)
+    b.add(1)
+
+    self.assertIn(1, b)
+    self.assertNotIn(2, b)
+
+  def test_not_compatible_different_seeds(self):
+    b1 = GeometricBloomFilter(length=15, random_seed=1)
+    b2 = GeometricBloomFilter(length=15, random_seed=2)
+    with self.assertRaises(AssertionError):
+      b1.assert_compatible(b2)
+    with self.assertRaises(AssertionError):
+      b2.assert_compatible(b1)
+
+  def test_not_compatible_different_lengths(self):
+    b1 = GeometricBloomFilter(length=10, random_seed=2)
+    b2 = GeometricBloomFilter(length=15, random_seed=2)
+    with self.assertRaises(AssertionError):
+      b1.assert_compatible(b2)
+    with self.assertRaises(AssertionError):
+      b2.assert_compatible(b1)
+
+  def test_compatible(self):
+    b1 = GeometricBloomFilter(length=15, random_seed=2)
+    b2 = GeometricBloomFilter(length=15, random_seed=2)
+    self.assertTrue(b1.assert_compatible(b2))
+    self.assertTrue(b2.assert_compatible(b1))
+
+
```

Comment:
Please also add a test in  FirstMomentEstimatorTest (line164) for GeometricBloomFitler.